modules/social_features/social_follow_content/social_follow_content.module

```diff
@@ -27,25 +27,27 @@ function social_follow_content_event_enrollment_update(EntityInterface $entity)
  */
 function social_follow_content_event_enrollment_follow(EntityInterface $entity) {
   if ($entity->field_enrollment_status->value) {
-    // Check if user already follow this content.
+    // Check if user is not anonymous and if user already follow this content.
     $account = \Drupal::currentUser();
-    $properties = [
-      'uid' => $account->id(),
-      'entity_type' => 'node',
-      'entity_id' => $entity->field_event->target_id,
-      'flag_id' => 'follow_content',
-    ];
-    $flaggings = \Drupal::entityTypeManager()
-      ->getStorage('flagging')
-      ->loadByProperties($properties);
-    $flagging = reset($flaggings);
+    if (!$account->isAnonymous()) {
```

Comment:
Can we add this as a guard clause at the top. i.e.

```
// Only logged in users can follow content.
if (!$account->isLoggedIn()) {
  return;
}
```

This avoids rewriting this entire function in this PR and communicates intent.

Reply:
It was fixed.